### PR TITLE
[Leo]: Use the same strategy for icons everywhere in Polymer

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/31772

**Note:** This PR won't actually bump Leo, but it depends on https://github.com/brave/leo/pull/337 landing in Leo before it can merge.

Currently, we have two different ways of overriding icons in Polymer. Each new page/component requires new overrides. This PR introduces an `iron-icon` override which let's us pass a Leo icon into an iron icon and replaces the two old methods with this.

In addition, it overrides a number of additional Chromium icons, which produces a much more consistent look & feel in the settings pages.

## New Leo Icons
- [x] downloads-off
- [x] motion-sensors-off
- [x] protocol-handler-on
- [x] protocol-handler-off
- [x] media-visualizer-off
- [x] usb-off
- [x] file-edit-off
- [x] sparkles-off
- [x] copy-off
- [x] credit-card-off
- [x] bluetooth-off
- [x] virtual-reality-off
- [x] windows-open-off
- [x] font-size-off
- [x] cookie-off
- [x] code-off
- [x] image-off
- [x] launch-off
- [x] open-in-browser
- [x] pdf
- [x] key-lock-off
- [x] user-off
- [x] smartphone-desktop-off

cc @aguscruiz for the new icons :smile:

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by aguscruiz** [comment]
All done with the same names except:

- File-edit-off is `file-corrupted`
- pdf we can just use `file`
- open-in-browser use `window`

**Comment by fallaciousreasoning** [comment]
Thanks to @aguscruiz for all the sweet new icons!

**Review by simonhong — COMMENTED** [review]
This PR looks great for me. but I'm not the polymer expert.
@petemill will give great review :)

File: browser/resources/settings/brave_overrides/iron_icon.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+RegisterPolymerPrototypeModification({
+  'iron-icon': (prototype) => {
+    const _updateIcon = prototype._updateIcon
+    prototype._updateIcon = function (...args: any[]) {
+      const removeAllOfType = (type: string) => {
+        for (const node of this.shadowRoot.querySelectorAll(type)) node.remove()
+      }
+
+      const name = iconMap[this.icon];
```
Comment (diracdeltas) [inline browser/resources/settings/brave_overrides/iron_icon.ts:91]: extra semicolon?

**Review by petemill — APPROVED** [review]
This is just the vision we had when we came up with Leo 👍

**Pull request merged**